core/src/main/java/org/infinispan/container/offheap/OffHeapDataContainer.java

```diff
@@ -39,7 +40,7 @@
  * @since 9.0
  */
 public class OffHeapDataContainer implements DataContainer<WrappedBytes, WrappedBytes> {
-   protected final Log log = LogFactory.getLog(getClass());
+   protected static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
```

Comment:
Unfortunately before this was used to show trace messages from the appropriate implementation. I have had to fix this before and the way to do this is to instead add an abstract `getLog` method to the abstract class that is implemented by the extended classes to properly say where the messages are from. This way the extending class can still define the logger as static but the abstract class uses the appropriate logger.

Reply:
Check out https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/stream/impl/AbstractCacheStream.java#L153